From ce4af248b0069f5e0bf4f5998ce67a06f97edd1b Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Wed, 4 Oct 2017 17:36:16 +0200
Subject: [PATCH] CAPI-538 test-ufds: delete customer test randomly failing
 with InvalidArgumentError Reviewed by: Patrick Mooney
 <patrick.mooney@joyent.com> Approved by: Patrick Mooney
 <patrick.mooney@joyent.com>

---
 test/capi/customers.test.js | 39 +++++++++++++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/test/capi/customers.test.js b/test/capi/customers.test.js
index 6ba4ca4..f8f717b 100644
--- a/test/capi/customers.test.js
+++ b/test/capi/customers.test.js
@@ -790,6 +790,45 @@ test('delete key', function (t) {
     });
 });
 
+// Given ufds-napi-watcher watches ufds changelog for new users creation
+// and adds a dclocalconfig to such users, we're gonna wait for this object
+// to appear, delete it, and move ahead to customer cleanup
+test('cleanup customer dclocalconfig', function (t) {
+    var counter = 0;
+    var limit = 60;
+    var localCfgDn = util.format(
+            'dclocalconfig=%s, uuid=%s, ou=users, %s',
+            CONFIG.datacenter_name, CUSTOMER.uuid, SUFFIX);
+
+    helper.createClient(function (clientErr, client) {
+        t.ifError(clientErr);
+        t.ok(client);
+
+        function _waitForDcLocalCfg() {
+            client.del(localCfgDn, function (err) {
+                if (err) {
+                    counter += 1;
+                    if (counter < limit) {
+                        setTimeout(_waitForDcLocalCfg, 5000);
+                    } else {
+                        t.ifError(err);
+                        client.unbind(function (err2) {
+                            t.ifError(err2);
+                            t.end();
+                        });
+                    }
+                } else {
+                    client.unbind(function (err2) {
+                        t.ifError(err2);
+                        t.end();
+                    });
+                }
+            });
+        }
+        _waitForDcLocalCfg();
+    });
+});
+
 
 test('delete customer', function (t) {
     CAPI.del('/customers/' + CUSTOMER.uuid, function (err, req, res) {
-- 
2.21.0

